### ISSUE TITLE
Pixelated image when cropping

### ISSUE THREAD

**Issue by dareniott** [jdamcd/android-crop#220]
When using Crop.of(URI_TO_SRC, URI_TO_OUTPUT).asSquare().start(getActivity(), this); to crop the attached source file (downloaded from the internet) the resulting image is strongly pixelated or downsampled.  Other built in image tools (Google Photos) for example crops it fine.  If another utility is used to modify the file and re-save it first.  Then when using the resulting (MODIFIED) image this crop library will properly crop the image without distortion.  Attached are the original source image exactly as it was downloaded and a modified version created simply by rotating the image and saving it through Google Photos.  I have narrowed the issue down to the CropImageActivity.decodeRegionCrop. This occurs on API23.

![src](https://cloud.githubusercontent.com/assets/2248325/14542392/b684bc4e-025c-11e6-8674-919909e39f22.jpg)
![rotated](https://cloud.githubusercontent.com/assets/2248325/14542391/b684a02e-025c-11e6-9f08-07d9177be1f4.jpg)

**Comment by dareniott** [comment]
Setting a breakpoint in CropImageActivity.saveOutput you can see that after the imageView is updated with the cropped image the resulting bitmap is distorted, 
![screenshot_20160418-081330](https://cloud.githubusercontent.com/assets/2248325/14603975/1f43ad6e-053f-11e6-9e41-4ddc1ac469a5.png)

while the crop of the same image having been modified (only by rotation) in Google Photos is not.

![screenshot_20160418-081350](https://cloud.githubusercontent.com/assets/2248325/14603976/230d8820-053f-11e6-8626-7f30eb725f95.png)

**Comment by Lawgrin** [comment]
I do believe that it is the image itself. if you check the size of the 2 files there is +- 10kb diff, from first glance it wouldn't seem that weird. if you place both images side by side the colour of the first image is lighter. also if you open the first image in Gallery (samsung) it distorts even with out cropping.

So imho try get a better quality image

**Comment by dareniott** [comment]
I agree that it is an issue with the image itself.  However on the Nexus devices the stock photo app handles cropping it fine which suggests that whatever the issue with the image, it is possible to work with the image without distorting it.  FYI - If you first open this ORIGINAL file as a bitmap (BitmapFactory.decodeStream) and then output a cached copy of it (bmp.compress(CompressFormat.JPEG, 100, outFile)) and pass that URI to the crop library instead of the the original file URI then the crop completes without issue.  Just shouldn't exactly be necessary to re-interpret the bmp and cache it somewhere.
